src/application/communication/Broker.java

```diff
@@ -227,22 +244,52 @@ public void send(byte[] buf)
     }
 
     /**
-     * Creates a bullet and returns it Id.
-     * @return The id of the bullet created.
+     * Sends a data with Tcp, as a byte array, to the server.
+     * 
+     * @param buf
+     *            The byte array to be sent.
      */
-    public int requestBulletCreation(double x, double y, double radius, double damage, int ownerId, int gameId)
+    public void sendTcp(byte[] data)
     {
         try
         {
-            return webservice.requestBulletCreation(x, y,radius,damage,ownerId,gameId);
+            output.write(data);
         }
-        catch (RemoteException e)
+        catch (IOException e)
         {
             e.printStackTrace();
-            return -1;
         }
-        
     }
+
+    /**
+     * Creates a bullet and returns it Id.
+     * 
+     * @return The id of the bullet created.
+     */
+    public void requestBulletCreation(GameObjectDAO data)
+    {
+        ByteBuffer buffer = ByteBuffer.allocate(256);
+
+        buffer.order(ByteOrder.LITTLE_ENDIAN);
+        buffer.put((byte)1); // ASCII Standard for Start of heading
+        buffer.put((byte)Opcodes.REQUEST_BULLET.getValue());
```

Comment:
ordinal returns its offset in the enum, not the specific value that we provie it in the private constructor we've made.